### PR TITLE
IARCRP-69: Make create_site async

### PR DESCRIPTION
**Pull request by dmitriiminaev**
[IARCRP-69](https://spherical.atlassian.net/browse/IARCRP-69)

### CODE REVIEW

**Review by mjecov — APPROVED** [review]

File: src/lektorium/repo/local/repo.py [hunk 1]
```diff
@@ -155,9 +155,9 @@ def unpark_session(self, session_id):
         session['edit_url'] = self.server.serve_lektor(session_dir)
         session.pop('parked_time', None)
 
-    def create_site(self, site_id, name, owner=None):
+    async def create_site(self, site_id, name, owner=None):
```
Comment (apatrushev) [inline src/lektorium/repo/local/repo.py:158]: serve_static below can take a long time, please at least create follow up issue

File: src/lektorium/repo/local/storage.py [hunk 2]
```diff
@@ -31,6 +32,13 @@
 run = functools.partial(subprocess.check_call, shell=True)
 
 
+def async_run(func, *args, **kwargs):
+    return get_event_loop().run_in_executor(
```
Comment (apatrushev) [inline src/lektorium/repo/local/storage.py:36]: Fine if it is working as expected. Could you please also create follow up to switch to asyncio way of running subprocesses?

File: src/lektorium/repo/local/storage.py [hunk 3]
```diff
@@ -140,7 +148,7 @@ def create_session(self, site_id, session_id, session_dir):
         site_root = self._site_dir(site_id)
         shutil.copytree(site_root, session_dir)
 
-    def create_site(self, lektor, name, owner, site_id):
+    async def create_site(self, lektor, name, owner, site_id):
```
Comment (apatrushev) [inline src/lektorium/repo/local/storage.py:151]: lektor.quickstart should be also async?

File: src/lektorium/repo/local/storage.py [hunk 4]
```diff
@@ -555,7 +563,7 @@ def create_site(self, lektor, name, owner, site_id):
 
         return site_workdir, options
 
-    def create_site_repo(self, site_id):
+    async def create_site_repo(self, site_id):
```
Comment (apatrushev) [inline src/lektorium/repo/local/storage.py:566]: init_project should also be async

Comment (dmitriiminaev) [reply src/lektorium/repo/local/storage.py:566]: I intended to work on Gitlab and AWS in a related [issue](https://spherical.atlassian.net/browse/IARCRP-88)

**Pull request merged**